sunpy/coordinates/tests/test_transformations.py

```diff
@@ -667,3 +669,32 @@ def test_gei_gei():
     new = old.transform_to(GeocentricEarthEquatorial(equinox=t, obstime=t)).cartesian
 
     assert_quantity_allclose(new.xyz, gei_d.xyz)
+
+
+def test_no_observer():
```

Comment:
I think it is worth testing here two different coordinate objects initialised with exactly the same parameters (but that are distinct objects) cannot be transformed, ie.:

```python
frame1 = Heliocentric(0*u.km, 0*u.km, 0*u.km, observer=None)
frame2 = Heliocentric(0*u.km, 0*u.km, 0*u.km, observer=None)
frame1.transform_to(frame2)   # This should fail
```

Reply:
As discussed on the telecon, this will actually succeed in the transformation machinery.